File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/utpeking/UtpekingsperioderDto.java
package no.nav.melosys.tjenester.gui.dto.utpeking;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

import no.nav.melosys.domain.Utpekingsperiode;

public record UtpekingsperioderDto(List<UtpekingsperiodeDto> utpekingsperioder) {
    public static UtpekingsperioderDto av(Collection<Utpekingsperiode> utpekingsperioder) {
        return new UtpekingsperioderDto(
            utpekingsperioder.stream()
                .map(UtpekingsperiodeDto::av)
                .collect(Collectors.toList())
        );
    }

    public static List<Utpekingsperiode> tilDomene(UtpekingsperioderDto utpekingsperioderDto) {
        return utpekingsperioderDto.utpekingsperioder.stream()
            .map(UtpekingsperiodeDto::tilDomene)
            .collect(Collectors.toList());
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/utpeking/UtpekingsperioderDto.kt
package no.nav.melosys.tjenester.gui.dto.utpeking

import no.nav.melosys.domain.Utpekingsperiode
import java.util.stream.Collectors

@JvmRecord
data class UtpekingsperioderDto(val utpekingsperioder: List<UtpekingsperiodeDto>) {
    companion object {
        @JvmStatic
        fun av(utpekingsperioder: Collection<Utpekingsperiode>): UtpekingsperioderDto {
            return UtpekingsperioderDto(
                utpekingsperioder.stream()
                    .map { utpekingsperiode: Utpekingsperiode -> UtpekingsperiodeDto.av(utpekingsperiode) }
                    .collect(Collectors.toList())
            )
        }

        @JvmStatic
        fun tilDomene(utpekingsperioderDto: UtpekingsperioderDto): List<Utpekingsperiode> {
            return utpekingsperioderDto.utpekingsperioder.stream()
                .map { obj: UtpekingsperiodeDto -> obj.tilDomene() }
                .collect(Collectors.toList())
        }
    }
}
